### ISSUE TITLE
Improve boot up speed

### ISSUE THREAD

**Issue by xiaohanyu** [xiaohanyu/oh-my-emacs#27]
The boot up speed is quite slow as there're more and more modules. Currently, My emacs takes about haft a minute just to boot up, so there should be some method to improve the bootup speed. 

Maybe https://github.com/jwiegley/use-package is a good reference.

**Comment by Plypy** [comment]
You could use the --daemon option

**Comment by xiaohanyu** [comment]
@Plypy Yeah I know that. I put a `alias ecc=emacsclient -c` to my `.bashrc` file

**Comment by xuchunyang** [comment]
What do you think about [use-package](https://github.com/joostkremers/visual-fill-column)? One of its purposes is to improve performance.

**Comment by xiaohanyu** [comment]
Hey, @xuchunyang : 

Thanks for you suggestion. I've checked use-package and I think it is not a good choice for oh-my-emacs right now. 

1. oh-my-emacs is built on top of `el-get` while use-package is built on top of melpa.
2. Generally, el-get provide ready-to-use info docs for its packages when available, while elpa just copy existing emacs lisp code to specified directory.
3. Uses can make their own recipes in el-get quite easily.

The boot up speed of emacs is always a problem for a long time in emacs community, that's why oh-my-emacs provide `core` features and optional `modules`.

Today I've figured out a simple idea to improve oh-my-emacs boot up speed further more.
1. Disable all `modules` when boot up for the first time.
2. When user open a file, say a ruby file for the first time, do `(ome-load "modules/ome-ruby")`, load when necessary, something like lazy evaluation.
